# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/base/test/test_text-fragments-create-text-directive.html
# Commit: 0232a68e3e77
# Full Hash: 0232a68e3e772c14fe2a3e5d5092e53ca660927e
# Author: Jan-Niklas Jaeschke <jjaschke@mozilla.com>
# Date: 2025-07-16 21:45:35
# Description:
#   Bug 1971670 - Text Fragments: Correctly deal with whitespace-only ranges. r=dom-core,avandolder
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D257502
# ==============================================================================

diff -r 9ae8f02369d2 -r 0232a68e3e77 dom/base/test/test_text-fragments-create-text-directive.html
--- a/dom/base/test/test_text-fragments-create-text-directive.html	Wed Jul 16 15:45:50 2025 +0000
+++ b/dom/base/test/test_text-fragments-create-text-directive.html	Wed Jul 16 15:49:29 2025 +0000
@@ -49,6 +49,7 @@
     <!-- empty text node will be inserted right here -->
     <span id="afterEmptyTextNode">after empty text node</span>
   </p>
+  <div id="nbsp">&nbsp;</div>
   <script>
     document.addEventListener("DOMContentLoaded", () => {
       const empty = document.createTextNode("");
@@ -428,6 +429,12 @@
         "Input range contains image only: Produces empty text directive"
       );
 
+      range.selectNode(nbsp);
+      textDirective = await SpecialPowers.wrap(document).fragmentDirective.createTextDirectiveForSelection();
+      is(
+        textDirective, null,
+        "Input range contains nbsp only: Produces empty text directive"
+      );
     }
 
     async function testExpandRangeToWordBoundaries() {
